components/ScrollToTop/index.js

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+import { styled } from "styled-components";
+
+const ScrollToTopElement = styled.div`
+  position: fixed;
+  bottom: 10px;
+  left: calc(50% - 25px);
+  font-size: 1.5em;
+  width: 50px;
+  text-align: center;
+  z-index: 999;
+  cursor: pointer;
+  color: var(--primaryColor);
+`;
+
+export default function ScrollToTop() {
+  const [visible, setVisible] = useState(false);
+  function toggleVisible() {
+    const scrolled = document.documentElement.scrollTop;
+    if (scrolled > 300) {
+      setVisible(true);
+    } else if (scrolled <= 300) {
+      setVisible(false);
+    }
+  }
+
+  function scrollToTop() {
+    if (typeof window !== "undefined") {
+      window.scrollTo({
+        top: 0,
+        behavior: "smooth",
+      });
+    }
+  }
+  if (typeof window !== "undefined") {
+    window.addEventListener("scroll", toggleVisible);
+  }
```

Comment:
You want to use the useEffect hook here. Whenever you attach an event listener, you need to make sure to remove that listener when the component unmounts. If you recall, you can use the `return` within the callback function to clean up any event listeners by returning the clean-up function as a callback.

Reply:
i will try it with useEffect :) Thanks!